### ISSUE TITLE
FreeBSD support

### ISSUE THREAD

**Issue by emaste** [flipperdevices/qFlipper#68]
Attempting to build on FreeBSD:
```
../../backend/flipperzero/protobufsession.cpp:448:2: error: "Unsupported OS"
#error "Unsupported OS"
 ^
1 error generated.
```
It's relatively easy to get past this and have qFlipper connect to the device (and I can put a pull request together).

However, it then appears the install process requires a per-host prebuilt binary, e.g. _https://update.flipperzero.one/builds/qFlipper/1.0.2/qFlipper-x86_64-1.0.2.AppImage_ so perhaps this will be a nonstarter?

**Comment by gsurkov** [comment]
We have chosen the AppImage bundle as a simple method of distributing the application on Linux systems (as we don't have resources or maintainers to build and support distro-specific packages yet).

In case you are willing to add FreeBSD support it is possible to make necessary changes to the build process, and then either submit a port into the [contrib](https://github.com/flipperdevices/qFlipper/tree/dev/contrib) section OR (preferred) maintain a separate repo with the files necessary to build a FreeBSD package.

In short: AppImage is not mandatory to build qFlipper, it can be packaged/distributed in any OS-native way.

**Comment by zhovner** [comment]
@emaste do you have any plans on this or can we close this issue?

**Comment by gsurkov** [comment]
Closing the issue. Please feel free to reopen it or create a new one if you have found a bug/have feature requests.

**Comment by h-2** [comment]
I am trying to get this to build on FreeBSD and it is not working for me. 

> It's relatively easy to get past this and have qFlipper connect to the device (and I can put a pull request together).

What did you do? I already get:

```
% make
cd 3rdparty/ && ( test -e Makefile || /usr/local/bin/qmake-qt5 -o Makefile /usr/home/hannes/devel/qFlipper/3rdparty/3rdparty.pro -spec linux-g++ CONFIG+=qtquickcompiler ) && make -f Makefile
make[1]: don't know how to make nanopb/pb_common.c. Stop
```

> In short: AppImage is not mandatory to build qFlipper, it can be packaged/distributed in any OS-native way.

It would be great if it was possible to just build the source-code on FreeBSD even if you do not provide official binaries.

Right now the situation is kinda frustrating. My desktop cannot build the app, so I cannot firmware-update. But the Android app also cannot perform a firmware-update, because it says the firmware on the device is too old (that's why I want to update it!!).

 😿
